Source/WebCore/style/AnchorPositionEvaluator.h

```diff
@@ -42,16 +45,61 @@ namespace WebCore {
 
 class Document;
 class Element;
+class LayoutPoint;
 class LayoutRect;
 class LayoutSize;
 class RenderBlock;
 class RenderBox;
 class RenderBoxModelObject;
 class RenderElement;
 class RenderStyle;
+class RenderView;
 
 enum CSSPropertyID : uint16_t;
 
+struct AnchorScrollSnapshot {
+    const RenderBox* m_scroller;
+    LayoutPoint m_scrollSnapshot { };
+    inline LayoutSize adjustmentForCurrentScrollPosition() const;
+};
+
+class AnchorScrollAdjuster {
+private:
+    LayoutSize adjustmentForViewport(const RenderView&) const;
+
+    RenderBox* m_anchored;
+    Vector<AnchorScrollSnapshot, 1> m_scrollSnapshots;
+    bool m_needsXAdjustment : 1 { false };
+    bool m_needsYAdjustment : 1 { false };
+    bool m_adjustForViewport : 1 { false };
+    bool m_hasChainedAnchor : 1 { false };
+    bool m_hasStickyAnchor : 1 { false };
+    bool m_isHidden : 1 { false };
+    bool m_hasFallback : 1 { false };
+    LayoutSize m_stickySnapshot;
+    LayoutSizeLimits m_fallbackLimits;
+
+public:
+    AnchorScrollAdjuster(RenderBox& anchored, const RenderBoxModelObject& defaultAnchor);
+    RenderBox* anchored() const { return m_anchored; }
+
+    bool mayNeedAdjustment() const { return m_needsXAdjustment | m_needsYAdjustment; }
+    inline bool isEmpty() const;
+    bool isHidden() const { return m_isHidden; }
+    void setHidden(bool hide) { m_isHidden = hide; }
+
+    inline void addSnapshot(const RenderBox& scroller);
+    inline void addViewportSnapshot(const RenderView&);
+
+    void setFallbackLimits(const RenderBox& anchored);
```

Comment:
Weird to have a "set" that doesn't take a limits argument. Maybe `computeFallbackLimits()`